src/main/java/com/ita/if103java/ims/service/impl/ItemServiceImpl.java

```diff
@@ -11,33 +11,39 @@
 import com.ita.if103java.ims.dto.SavedItemDto;
 import com.ita.if103java.ims.dto.WarehouseDto;
 
-
+import com.ita.if103java.ims.entity.Event;
+import com.ita.if103java.ims.entity.EventName;
+import com.ita.if103java.ims.entity.Item;
 import com.ita.if103java.ims.entity.SavedItem;
 import com.ita.if103java.ims.entity.Transaction;
 import com.ita.if103java.ims.entity.TransactionType;
-import com.ita.if103java.ims.entity.User;
-import com.ita.if103java.ims.entity.Event;
-import com.ita.if103java.ims.entity.EventName;
 import com.ita.if103java.ims.entity.Warehouse;
 import com.ita.if103java.ims.exception.ItemNotEnoughCapacityInWarehouseException;
 import com.ita.if103java.ims.exception.ItemNotEnoughQuantityException;
+import com.ita.if103java.ims.exception.ItemNotFoundException;
+import com.ita.if103java.ims.exception.SavedItemNotFoundException;
 import com.ita.if103java.ims.mapper.ItemDtoMapper;
 import com.ita.if103java.ims.mapper.SavedItemDtoMapper;
 import com.ita.if103java.ims.mapper.WarehouseDtoMapper;
+import com.ita.if103java.ims.security.UserDetailsImpl;
 import com.ita.if103java.ims.service.EventService;
 import com.ita.if103java.ims.service.ItemService;
 import com.ita.if103java.ims.util.ItemTransactionUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Service;
 
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
 @Service
 public class ItemServiceImpl implements ItemService {
```

Comment:
Please format this class, because It is hard to review.
Line width should be 120 characters